core/crypto/pedersen_hash.go

```diff
@@ -11,17 +9,15 @@ import (
 //
 // [Pedersen array hashing]: https://docs.starknet.io/documentation/develop/Hashing/hash-functions/#pedersen_hash
 func PedersenArray(elems ...*felt.Felt) (*felt.Felt, error) {
-	if len(elems) < 3 {
-		return nil, errors.New("number of elems must be more than 2")
-	}
-
 	var err error
 	d := new(felt.Felt).SetZero()
 
-	for _, e := range elems {
-		d, err = Pedersen(d, e)
-		if err != nil {
-			return nil, err
+	if len(elems) > 0 {
```

Comment:
Do we need this check?

Reply:
I think I added it cos I was getting some error on arrays of length 0. Will cross-check again